### DISCUSSION TITLE
Custom Socialite Provider

### DISCUSSION THREAD

**Discussion by ARCHES-Academy** [statamic/cms#6355] [Q&A]
I'm trying to configure a custom SSO with Clever, which is an SSO provider for schools. I've successfully set up Socialite before using Google, but a Clever integration doesn't currently exist for Socialite (that I can find). The [statamic docs](https://statamic.dev/oauth#custom-providers) mention how to create a custom provider in passing, but don't even get close to giving the actual instructions for how to do so. As a result, I am very confused.

I've gone through it, and the `CleverExtendSocialite.php` is pretty straightforward (I think):
```php
<?php

use SocialiteProviders\Manager\SocialiteWasCalled;

class CleverExtendSocialite
{
    /**
     * Register the provider.
     *
     * @param \SocialiteProviders\Manager\SocialiteWasCalled $socialiteWasCalled
     */
    public function handle(SocialiteWasCalled $socialiteWasCalled)
    {
        $socialiteWasCalled->extendSocialite('clever', Provider::class);
    }
}
```

Now, the other file, `Provider.php`, I'm so lost on. I've looked through the GitHub repos for the various providers, and there seems to be so much variance between them.

I guess I just need a direction to go. What information is needed, and where do I put it? I have my Clever API credentials, and I'm sure they go in there somewhere, but I don't think they go in the `Provider.php`. Auth & Redirect URLs? What else am I missing?

**Comment by ryanmitchell** [comment]
Here's an example of a Provider.php file I found via https://socialiteproviders.com/Acclaim/: https://github.com/SocialiteProviders/Acclaim/blob/master/Provider.php

Does that help you understand whats required? The Statamic docs dont explain it as its a Laravel package...

**Reply by ARCHES-Academy** [reply]
I'd already looked at a few examples, and they weren't super helpful. I was hoping there would be some documentation available from this side since Statamic does have a few unique quirks in regards to login, but I guess I'll have to get in touch with some folks over at the Socialite Providers team and the Clever team. Thanks!

**Comment by leenamac** [comment]
Single Sign-On does have some drawbacks. For example, you might have applications that you want to have locked down a bit more. For this reason, it would be important to choose an SSO solution that gives you the ability to, say, require an additional authentication factor before a user logs into a particular application or that prevents users from accessing certain applications unless they are connected to a secure network. [https://sinuanonoche.org/](https://sinuanonoche.org/): [https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-saml-single-sign-on-for-your-organization/about-identity-and-access-management-with-saml-single-sign-on](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-saml-single-sign-on-for-your-organization/about-identity-and-access-management-with-saml-single-sign-on)
